tests/basic/replay.rs

```diff
@@ -133,3 +135,20 @@ fn replay_deadlock3_drop_mutex() {
     let runner = Runner::new(scheduler, Default::default());
     runner.run(deadlock_3);
 }
+
+// Check that FailurePersistence::None does not print a schedule
+#[test]
+fn replay_persist_none() {
+    let result = panic::catch_unwind(|| {
+        let scheduler = PctScheduler::new(2, 100);
+        let mut config = Config::new();
+        config.failure_persistence = FailurePersistence::None;
+        let runner = Runner::new(scheduler, config);
+        runner.run(concurrent_increment_buggy);
+    })
+    .expect_err("test should panic");
+    let output = result.downcast::<String>().unwrap();
+    assert!(output.contains("counter is wrong"));
+    // All our current failure persistence modes print the word "schedule", so check that's missing
```

Comment:
Was thinking we should record this assumption somewhere... though that's probably not a big deal as this test will catch a violation anyway.

Reply:
Yeah, I was thinking the same but wasn’t really sure how to do it. Even  testing this at all seemed like overkill anyway, so I think we’re fine with it being potentially a bit fragile.